tsnake/grid.py

```diff
@@ -467,18 +472,52 @@ def reparameterize_phase_one(self, snakes: [TSnake]) -> ([TSnake], deque([Point]
 
             new_snake = TSnake(nodes=new_nodes, force=self.image_force,
                                intensity=self.image_inflation_force, a=a, b=b, q=q, gamma=gamma, dt=dt)
-            # print('intersections:', intersections)
-            # Compute intersections in counter-clockwise direction,
-            # so subsequent normal calculation is performed correctly.
-            # Create new snake nodes in same order as intersections were computed.
-            # Initialize snake in same order as intersections.
             new_snakes.append(new_snake)
+            grid_node_queues.append(grid_node_queue)
 
             # NOTE: Create new snake nodes in same order as intersections were computed.
             # Initialize snake in same order as intersections.
             # If the snake split, add the newly created snakes too.
 
-        return (new_snakes, grid_node_queue)
+        return (new_snakes, grid_node_queues)
+
+
+    def reparameterize_phase_two(self, snakes: [TSnake], grid_node_queues: [deque([Point])]) -> [TSnake]:
+        '''
+        Takes a list of ACID grid verticies found in reparameterize_phase_one()
+        and performs reparametrize phase two. It then returns a new list of T-Snakes (which have 
+        possibly been split/merged/etc). If no snakes are split/merged, then the snakes are 
+        presented in the same order as before, AND each snake's nodes are presented in the same order.
+        
+         Arguments:
+        ===================================
+        * grid_node_queues: list(deque([Point])) to be dequeued and processed, 1 for each
+        snake that was passed to reparametrize_phase_1
+        * snakes: List of the old Tsnakes
+        ===================================
+        Returns:
+        ===================================
+        * [Tsnake] is a list of the reparametrized TSnakes (same order if no splits or merges)
+        ===================================
+        '''
+        # Nodes that have just been turned on, i.e.
+        # new outside boundary of the snake
+        for grid_node_queue in grid_node_queues:
```

Comment:
@zimmerman-cole This is the flood fill, it isn't factored out to it's own function but that should be relatively trivial. The other piece you'll need to add if you use BFS like I do here is a notion of visited nodes - here I use the fact that a node is 'on' or 'off' to handle that. You could potentially re-use that as well, though you may need to allocate a simple 2D array to track that for the image/mask

Reply:
Hopefully, this is helpful, I realize that it doesn't generalize particularly well.

Reply:
So is it the nodes that are turned on that are inside the snake?

Reply:
Because our snake is shrinking the behavior is the opposite, nodes that are outside the snake are turned on. I couldn't do the reverse because of the "once turned on, stay on" property. But it works the same, the nodes that are off delineate the inside of the snake. Might be easiest to deal with the conflicts and merge this into master, and then just use reparametrzie phase II as the flood fill itself.

Reply:
Which I don't mind doing if you're working on other stuff to make sure main.py works, the other piece I'm not sure about in there is the halt conditions, with the implementation we have I'm not sure those will be met.

Reply:
Or is it that turned-on nodes are the endpoints of grid-cell-edges which intersected w/ the snake?

Reply:
Yeah exactly, and they are the seed for the flood fill to fill the new outside portion of the snake. I.e. they are the nodes inside the grid_node_queue when the while loop on line 508 starts.